### PR TITLE
Update Vis Docs

### LINKED ISSUE #337: Visualization questions and cleanup suggestions
Looking more closely at these new visualizations, and there are a few issues I'd like to see resolved.
1. This should be one line per **_canonical**_ move. Separating the minus move into multiple steps is just confusing. At the very least, there needs to be an option for this.
2. How do I turn off showing replica exchange? Or turn off showing minus moves? This is what we'd usually want to look at. And if, after a minus move, the next path might be completely decorrelated: that's fine. It should be like starting a new tree with the decorrelated segment of the minus trajectory.
3. What are the numbers to the left? Can I turn them off? The representation of the move should be labelled with the step number, and possibly a short string to identify move type (I mostly agree with your selections: "f", "b", "r", "x", and I'd use "-" for the minus move; possibly "f-"/"b-" to differentiate direction.) Other numbers are just confusing. I think the number you have next to the path is actually _not_ the step number -- step number is at the far left now, right? (All these differences from the behavior of old scripts will only confuse users -- let's keep things as similar as possible.)
4. How do I get this to show rejected moves? That was one of the most useful new aspects of this visualization in a previous version. Now I don't see any difference when I set `tree.rejected=True`.

I think you might be losing sight of the way we actually use these "path tree" visualizations. The purpose is _not_ to be a log of every substep of the move, tracking provenance. Rather, it gives important information on how paths are decorrelating. The current version of these includes a bunch of visual "extras" that distract from the point.

Examples of the kinds of information that we get from these pictures:
- _Does the path length oscillate between different characteristic lengths?_ This could indicate switching between two mechanisms.
- _Do we see blocks where one direction of shooting is consistently rejected?_ This tells us something about early-vs-late barriers in the system.
- _Do rejected shooting moves from a particular region have much longer trajectories that still end in the states?_ Again, this suggests multiple channels, and may indicate that we're completely missing one channel.

These are just a few sorts of things I've thought of in past while looking at these plots. That doesn't mean the list is exclusive. The point is that asymmetries in the tree can inspire us to ask questions about why. Those are easier to catch if the visualization is less information-dense.
---

As an aside, #335 includes a subclass for the move tree visualization in order to simplify for the common case. It makes the user's tasks (which are what the notebooks are to illustrate) much easier.

### PR DESCRIPTION
**Pull request by jhprinz**
Resolves #337 .

Just an update on what visualization can do. No code changes so merge at any time!

### CODE REVIEW

**Comment by dwhswenson** [comment]
This definitely looks like it'll be the sort of thing I was hoping for.

Could you add a version after running it? This will go straight into the documentation, but that's only useful if there's a version that has been run.

Also, did the `join_blocks` option (what I like to call the "boxcars" version, because the trajectory looks like a long train) get renamed to `horizontal_gap`? This is one that I like to use, especially for example code, so I want to know how to make it work!

**Comment by jhprinz** [comment]
Last thing: I looked into the wrong step numbers and turns out:

Step Zero / the initial step has `.change = EmptyMoveChange()`. An EmptyMoveChange is technically rejected for it does not replace samples and hence the initial step is rejected. 
Since its `.change.samples` property is empty it is interpreted as if the initial change did not create any samples and this is what is used to determine the (generating) step. Thus the `*` and not number 0. This also causes the initial step to be show as a rejected step, which is weird since this step does not have an accepting part.

I propose to use an `InitialSampleSetMoveChange` that has all the samples in it that are used and makes this consistent. I can tweak visualization to show old files correctly. Suggestions?

I have also added a new column for the legend to show the range of steps where this sample was in the active set. Where the `step` legend is actually the step in which the sample was produced. An example what confused me first and led to this option was that I used `steps[0:100]` and the last shown step number was `76`. This is because the sample was made in `76` but is active in `76-99`. 

 I still need to look into the question, whether `accepted=True/False` for ReplicaStepList returns the correct list, but other than that I am done.

**Comment by jhprinz** [comment]
> I propose to use an InitialSampleSetMoveChange that has all the samples in it that are used and makes this consistent. I can tweak visualization to show old files correctly. Suggestions?

Or we just use the `AcceptedSampleMoveChange` without a Mover. This will also give the correct result.

**Comment by dwhswenson** [comment]
> Or we just use the `AcceptedSampleMoveChange` without a Mover

This seems reasonable, but if you do this, double check that we're still getting the correct results out of `scheme.move_summary`. I think we will (because it [looks like the summary functions ignore changes with `mover=None`](https://github.com/openpathsampling/openpathsampling/blob/master/openpathsampling/high_level/move_scheme.py#L741)). Just be sure we don't get a weird extra count in there.

Changing the code that I linked above from `move_scheme.py` will have weird effects on SRTIS, so if you change that, you need to check the behavior of SRTIS analysis as well. Preferably, don't change it.

**Comment by jhprinz** [comment]
> Just be sure we don't get a weird extra count in there.

Yes, that should be fine. I did not need to change any of that analysis code. The mover of the initial change is `None` as it should be. And if you really restart there will be no additional step. Hence the counting will not change.

**Comment by dwhswenson** [comment]
@jhprinz: When you have a chance, could you merge `upstream/master` into this (`visdocs`), to remove the commits from the `restart` branch from the diffs to be reviewed here? Thanks!

File: openpathsampling/snapshot_modifier.py [hunk 1]
```diff
@@ -37,6 +38,7 @@ class SnapshotModifier(StorableNamedObject):
     __metaclass__ = abc.ABCMeta
 
     def __init__(self, subset_mask=None):
+        super(SnapshotModifier, self).__init__()
```
Comment (jhprinz) [inline openpathsampling/snapshot_modifier.py]: Actually I think this was missing. It is necessary for storable objects. In rare cases you might
not do it. Just to point @dwhswenson to a potential mistake.

**Pull request merged**